stackable-base/Dockerfile

```diff
@@ -5,6 +5,14 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:8.9@sha256:87bcbfedfd70e67aab38
 # intentionally unused
 ARG PRODUCT
 
+# Sets the default shell to Bash with strict error handling and robust pipeline processing.
+# "-e": Exits immediately if a command exits with a non-zero status
+# "-u": Treats unset variables as an error, preventing unexpected behavior from undefined variables.
+# "-o pipefail": Causes a pipeline to return the exit status of the last command in the pipe that failed, ensuring errors in any part of a pipeline are not ignored.
+# "-c": Allows the execution of commands passed as a string
+# This is automatically inherited in all other Dockerfiles that use this unless it is overwritten
+SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
```

Comment:
[hadolint](https://github.com/hadolint/hadolint) complains about the missing SHELL command in the other Dockerfiles:

```
airflow/Dockerfile:46 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
```

This warning should be ignored by adding the following line to all Dockerfiles which use stackable-base:

```Dockerfile
# hadolint global ignore=DL4006
```

Reply:
Thanks! I'll add that.

@razvan is it too late to get this into 24.3?

Reply:
This should be done.